clients/repo.go

```diff
@@ -16,6 +16,7 @@ package clients
 
 // Repo interface uniquely identifies a repo.
 type Repo interface {
+	Path() string
```

Comment:
does this need to be its own method or can the caller just call `URI()` and strip the `Host()` prefix?

Reply:
I think that's possible, but I'm wary of unexpectedly-shaped enterprise URLs or substring operations being finicky.

Reply:
SGTM.

Now that there are multiple, closely related parts, can you add some brief comments in the interface method so the godoc explains the differences?

```go
type Repo interface {
        // URI returns everything ... 
	URI() string
 	// Path returns ...
	Path() string
        // Host returns ...
	Host() string
	String() string
	IsValid() error
	Metadata() []string
	AppendMetadata(metadata ...string)
}
```